File: python/open_pyramid_pattern.py
#  open pyramid of size 5:
#      *
#     * *
#    *   *
#   *     *
#  * * * * *

n=int(input())
space=n-1
mid_space=1
for i in range(1,n):
	for j in range(space):
		print(" ",end='')				#spaces before
	if i==1:
		print("*",end='')              	#first layer
	else:                               #middle layers
		print("*",end='')
		for j in range(mid_space):
			print(" ",end='')
		print("*",end='')
		mid_space+=2
	space-=1
	print("\n",end='')
for i in range(n):
	print("* ",end='');						#last layer

File: java/open_pyramid_pattern.java
// open pyramid of size 5:
//     *
//    * *
//   *   *
//  *     *
// * * * * *

import java.util.Scanner;

class HelloWorld{
    public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n=sc.nextInt();
		int space=n-1;
		int mid_space=1;
		for(int i=1;i<n;i++){
			for(int j=0;j<space;j++) System.out.print(" "); //spaces before *
			if(i==1) System.out.print("*");                 //first layer
			else{                          			        //middle layers
				System.out.print("*");
				for(int j=0;j<mid_space;j++) System.out.print(" ");
				System.out.print("*");
				mid_space+=2;
			}
			space--;
			System.out.print("\n");
		}
		for(int i=0;i<n;i++) System.out.print("* ");	//last layer
	}
}